src/test/ts/alien/Loader.tsx

```diff
@@ -18,78 +23,80 @@ const getRoot = () => Optional.from(document.getElementById('root')).getOrThunk(
-        setTimeout(() => {
-          Optional.from(ref.current)
-            .map(ReactDOM.findDOMNode)
-            .bind(Optional.from)
-            .map(SugarElement.fromDom)
-            .filter(SugarNode.isHTMLElement)
-            .map((val) => val.dom)
-            .fold(() => die('Could not find DOMNode'), (DOMNode) => {
-              next({
-                ref,
-                editor,
-                DOMNode
+        editor.on('SkinLoaded', () => {
+          setTimeout(() => {
+            Optional.from(ref.current)
+              .map(ReactDOM.findDOMNode)
+              .bind(Optional.from)
+              .map(SugarElement.fromDom)
+              .filter(SugarNode.isHTMLElement)
+              .map((val) => val.dom)
+              .fold(() => reject('Could not find DOMNode'), (DOMNode) => {
+                resolve({
+                  ref,
+                  editor,
+                  DOMNode,
+                });
               });
-            });
-        }, 0);
-      });
-    }
-  };
+          }, 0);
+        });
+      }
+    };
 
-  /**
+    /**
      * NOTE: TinyMCE will manipulate the DOM directly and this may cause issues with React's virtual DOM getting
      * out of sync. The official fix for this is wrap everything (textarea + editor) in an element. As far as React
      * is concerned, the wrapper always only has a single child, thus ensuring that React doesn’t have a reason to
      * touch the nodes created by TinyMCE. Since this only seems to be an issue when rendering TinyMCE 4 directly
      * into a root and a fix would be a breaking change, let's just wrap the editor in a <div> here for now.
      */
-  ReactDOM.render(<div><Editor ref={ref} apiKey='no-api-key' {...props} init={init} /></div>, getRoot());
-});
+    ReactDOM.render(<div><Editor ref={ref} apiKey='no-api-key' {...props} init={init} /></div>, container);
+  });
 
-// By rendering the Editor into the same root, React will perform a diff and update.
-const cReRender = (props: Partial<IAllProps>) => Chain.op<Context>((context) => {
-  ReactDOM.render(<div><Editor apiKey='no-api-key' ref={context.ref} {...props} /></div>, getRoot());
-});
-
-const cRemove = Chain.op((_) => {
-  ReactDOM.unmountComponentAtNode(getRoot());
-});
-
-const cNamedChainDirect = (name: keyof Context) => NamedChain.direct(
-  NamedChain.inputName(),
-  Chain.mapper((res: Context) => res[name]),
-  name
-);
-
-const cDOMNode = (chain: Chain<Context['DOMNode'], unknown>): Chain<Context, Context> => NamedChain.asChain<Context>([
-  cNamedChainDirect('DOMNode'),
-  NamedChain.read('DOMNode', chain),
-  NamedChain.outputInput
-]);
-
-const cEditor = (chain: Chain<Context['editor'], unknown>): Chain<Context, Context> => NamedChain.asChain<Context>([
-  cNamedChainDirect('editor'),
-  NamedChain.read('editor', chain),
-  NamedChain.outputInput
-]);
+  const remove = () => {
+    ReactDOM.unmountComponentAtNode(container);
+    deleteTinymce();
```

Comment:
It would remove tinymce when the context leaves scope, but it will also simplify the tests so I'm leaning toward it being a good idea anyway.

Reply:
Yes but it's a new concept where the BDD mocha hooks are something we use everywhere else to do these kind of context based states.
Using new things might be cool but since this is so cutting edge that you need to monkey patch things to make it work I'm not so sure. I guess since this is scoped to just this project it might be worth evaluating if we should use disposables else ware.
In this case loading/unloading in each test is likely not a problem since there is so few tests but that would never work if you had like thousands of these things all time would be spent on just network traffic.

Reply:
I played around with keeping a string key (contained the version and other stringified editor props) that the loader would check first before calling `deleteTinymce`. It didn't seem to make a difference to performance though but I guessed it's just because of network cache on local testing.
I'm happy to change it to use miniature on the non load tests, but since the React integration can load TinyMCE on it's own I thought why not use that functionality on all tests.
Also to do with the eslint override, eslint v9 fixes the `using` keyword bug: [commit](https://github.com/eslint/eslint/commit/610c1486dc54a095667822113eb08062a1aad2b7). I can see there might be some migration needed there to use v9 though.

Reply:
Right maybe miniature is overkill also we no longer have a way to publish that library so maybe the integrations should use less our our libraries. But I wonder if we need to share things like this between integrations. Maybe worth discussing in tinymce-dev.

Reply:
Miniature _is_ that code sharing, more or less, we can still publish it in theory I just didn't have time to set it up. It could be published manually for now.

Reply:
I've moved the `deleteTinymce` call out of the `ReactEditorContext` completely. It's now back in `LoadTinyTest.ts`. The rest of the tests now use `Loader.withVersion` which internally uses miniature to handle loading core TinyMCE versions.